src/TypeScript/EslintBridgeClient/EslintBridgeStartUp.cs

```diff
@@ -19,17 +19,183 @@
  */
 
 using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Abstractions;
+using System.Text.RegularExpressions;
 using System.Threading.Tasks;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.Helpers;
+using SonarLint.VisualStudio.TypeScript.NodeJSLocator;
 
 namespace SonarLint.VisualStudio.TypeScript.EslintBridgeClient
 {
-    // TODO: https://github.com/SonarSource/sonarlint-visualstudio/issues/2256
-
     internal interface IEslintBridgeStartUp : IDisposable
     {
         /// <summary>
         /// Ensures that the process is running. Returns the port that server is running on.
         /// </summary>
         Task<int> Start();
     }
+
+    [Export(typeof(IEslintBridgeStartUp))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
```

Comment:
I'm not clear how the lifetime of the various objects is going to work. How do you see this class and the `EsLintBridgeHttpWrapper` being consumed by the analyzers?

How would we stop the node process when a solution is closed, or have separate processes for TS and JS?

Reply:
If we decide to go with 2 processes then this will have to be changed, probably to a Factory class (that is a singleton), that receives the dependencies in the ctor and news-up the instances.
Stopping when a solution is closed: I haven't exposed a Stop method yet, but it's something that we could do if we decide to terminate the process when a solution is closed.
Do you think that it's worth doing the changes now?

Reply:
Not now you've written it like this, but I think it's worth thinking about the overall UX and how it's going to work end-to-end with the analysers before doing much more to avoid unnecessary rework later.